app/src/org/commcare/android/models/logic/FormRecordProcessor.java

```diff
@@ -61,12 +63,20 @@ public FormRecord process(FormRecord record)
             FormUploadUtil.getDecryptCipher((new SecretKeySpec(record.getAesKey(), "AES")));
         InputStream is = new CipherInputStream(new FileInputStream(f), decrypter);
 
-        AndroidTransactionParserFactory factory = new AndroidTransactionParserFactory(c, null);
-
-        factory.initCaseParser();
-        factory.initStockParser();
+        AndroidTransactionParserFactory factory = new AndroidTransactionParserFactory(c, null) {
+            @Override
+            public TransactionParser getParser(KXmlParser parser) {
+                String namespace = parser.getNamespace();
+                String name = parser.getName();
+                if (LedgerXmlParsers.STOCK_XML_NAMESPACE.equals(namespace) || "case".equalsIgnoreCase(name)) {
```

Comment:
The fix is to use `AndroidTransactionParserFactory` parse logic, but limit it only to those parsers that were used before these changes: https://github.com/dimagi/commcare/pull/224/files#diff-f43ca72053194678359963951db09445